framework/src/utils/RandomData.C

```diff
@@ -96,32 +96,29 @@ RandomData::updateGenerators()
   }
 
   if (_is_nodal)
-  {
-    MeshBase::const_node_iterator it = _rd_mesh.getMesh().active_nodes_begin();
-    MeshBase::const_node_iterator end_it = _rd_mesh.getMesh().active_nodes_end();
+    updateGeneratorHelper(_rd_mesh.getMesh().active_nodes_begin(), _rd_mesh.getMesh().active_nodes_end());
```

Comment:
Ummm... then there is no way this can work with ParallelMesh... so the whole thing needs to be redesigned.

Having a serial loop over all elements / nodes is bad news anyway.  It should be threaded (which would also preclude this design).